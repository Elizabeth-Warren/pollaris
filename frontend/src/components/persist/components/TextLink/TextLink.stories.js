import React from 'react';
import { action } from '@storybook/addon-actions';
import { select, text } from '@storybook/addon-knobs';
import startCase from 'lodash/startCase';
import TextLink, { TextLinkLevels } from './index';

const handleClick = (e) => {
  e.preventDefault();
  action('link clicked');
};

export default {
  title: 'Components/TextLink',
  component: TextLink,
  parameters: { jest: ['TextLink'] },
};

export const withKnobs = () => {
  const linkText = text('Text', 'Click me');
  const level = select('Level', Object.values(TextLinkLevels), TextLink.defaultProps.level);
  return (<TextLink level={level} onClick={handleClick} href="#">{linkText}</TextLink>);
};

export const gallery = () => (
  <div>
    {Object.values(TextLinkLevels).map((level) => (
      <p key={level}>
        <TextLink level={level} onClick={handleClick} href="#">
          { startCase(level) }
          {' '}
          Link Style
        </TextLink>

      </p>
    ))}
  </div>
);
