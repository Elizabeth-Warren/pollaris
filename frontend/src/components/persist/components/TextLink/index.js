import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Chevron from '../Chevron';
import theme from '../../theme';

export const TextLinkLevels = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
};

const ChevronWrapper = styled.span`
  color: ${theme.colors.red};
  font-size: 1em;
  left: 0;
  position: relative;
  transition: all 150ms ease-in-out;
`;

const LinkWrapper = styled.a`
  background: ${({ level }) => (level === TextLinkLevels.SECONDARY
    ? `linear-gradient(to bottom,transparent 0%,transparent 65%,${theme.colors.liberty} 65%,${theme.colors.liberty} 100%)`
    : null)
};
  border-bottom: ${({ level }) => (level === TextLinkLevels.TERTIARY ? `2px solid ${theme.colors.liberty}` : 'none')};
  color: ${theme.colors.black};
  cursor: pointer;
  font-family: ${theme.fontFamily.sans};
  font-size: ${theme.fontSize.md};
  font-weight: bold;
  text-decoration: none;

  &:hover,
  &:focus {
    background: none;
    border-bottom: ${({ level }) => (level === TextLinkLevels.TERTIARY ? `2px solid ${theme.colors.black}` : 'none')};

    & ${ChevronWrapper} {
      left: 0.5em;
    }
  }

  &:active {
    background: none;
    border-bottom: ${({ level }) => (level === TextLinkLevels.SECONDARY ? `2px solid ${theme.colors.liberty}` : 'none')};
  }
`;

const TextLink = ({ children, level, ...other }) => (
  <LinkWrapper level={level} {...other}>
    {children}
    {level === TextLinkLevels.PRIMARY && ' '}
    {level === TextLinkLevels.PRIMARY && <ChevronWrapper as={Chevron} />}
  </LinkWrapper>
);

TextLink.propTypes = {
  /** Child text or elements */
  children: PropTypes.node.isRequired,
  /** Determines button color. One of 'primary', 'secondary', or 'tertiary'. */
  level: PropTypes.oneOf(Object.values(TextLinkLevels)),
};

TextLink.defaultProps = {
  level: TextLinkLevels.PRIMARY,
};

export default TextLink;
