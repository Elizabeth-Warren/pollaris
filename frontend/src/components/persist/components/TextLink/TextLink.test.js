import React from 'react';
import renderer from 'react-test-renderer';
import startCase from 'lodash/startCase';
import TextLink, { TextLinkLevels } from './index';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <div>
        {Object.values(TextLinkLevels).map((level) => (
          <p key={level}>
            <TextLink level={level} href="#">
              { startCase(level) }
              {' '}
              Link Style
            </TextLink>

          </p>
        ))}
      </div>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
