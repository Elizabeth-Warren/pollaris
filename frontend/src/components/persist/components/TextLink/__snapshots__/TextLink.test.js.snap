// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  border-color: currentColor;
  border-style: solid;
  border-width: 0.15em 0.15em 0 0;
  box-sizing: border-box;
  display: inline-block;
  height: 0.45em;
  left: -0.15em;
  position: relative;
  top: -0.1em;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  width: 0.45em;
}

.c3 {
  color: #b61b28;
  font-size: 1em;
  left: 0;
  position: relative;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
}

.c0 {
  border-bottom: none;
  color: #171717;
  cursor: pointer;
  font-family: 'Ringside Regular A','Ringside Regular B','Helvetica',sans-serif;
  font-size: 18px;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  background: none;
  border-bottom: none;
}

.c0:hover .c2,
.c0:focus .c2 {
  left: 0.5em;
}

.c0:active {
  background: none;
  border-bottom: none;
}

.c4 {
  background: linear-gradient(to bottom,transparent 0%,transparent 65%,#b7e4cf 65%,#b7e4cf 100%);
  border-bottom: none;
  color: #171717;
  cursor: pointer;
  font-family: 'Ringside Regular A','Ringside Regular B','Helvetica',sans-serif;
  font-size: 18px;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover,
.c4:focus {
  background: none;
  border-bottom: none;
}

.c4:hover .c2,
.c4:focus .c2 {
  left: 0.5em;
}

.c4:active {
  background: none;
  border-bottom: 2px solid #b7e4cf;
}

.c5 {
  border-bottom: 2px solid #b7e4cf;
  color: #171717;
  cursor: pointer;
  font-family: 'Ringside Regular A','Ringside Regular B','Helvetica',sans-serif;
  font-size: 18px;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover,
.c5:focus {
  background: none;
  border-bottom: 2px solid #171717;
}

.c5:hover .c2,
.c5:focus .c2 {
  left: 0.5em;
}

.c5:active {
  background: none;
  border-bottom: none;
}

<div>
  <p>
    <a
      className="c0"
      href="#"
    >
      Primary
       
      Link Style
       
      <span
        className="c1 c2 c3"
        color="currentColor"
        direction="right"
      />
    </a>
  </p>
  <p>
    <a
      className="c4"
      href="#"
    >
      Secondary
       
      Link Style
    </a>
  </p>
  <p>
    <a
      className="c5"
      href="#"
    >
      Tertiary
       
      Link Style
    </a>
  </p>
</div>
`;
